prov/tcp/src/tcpx_comm.c

```diff
@@ -88,7 +88,16 @@ static int tcpx_process_read_rsp(struct tcpx_pe_entry *pe_entry)
 
 static int tcpx_process_write(struct tcpx_pe_entry *pe_entry)
 {
-	return -FI_ENODATA;
+	int i;
+
+	pe_entry->msg_data.iov_cnt = pe_entry->rma_data.rma_iov_cnt;
+	for ( i = 0 ; i < pe_entry->rma_data.rma_iov_cnt ; i++ ) {
+		pe_entry->msg_data.iov[i].iov_base =
+			(void *) pe_entry->rma_data.rma_iov[i].addr;
+		pe_entry->msg_data.iov[i].iov_len =
+			pe_entry->rma_data.rma_iov[i].len;
+	}
```

Comment:
Is this iov copy really necessary?  We're copying from within the same pe_entry structure.

On this note, the name 'pe_entry' seems confusing.  Maybe we should just have tx/rx_entry or tx/rx_req or tx/rx_something...  PE stood for progress engine in the socket provider, but we really don't have a progress engine here, just structures to track outstanding transmits or incoming receives.

Reply:
This is on the receive side after getting the rma data, it is validated and copied to iov such that the readv can be called on socket directly to get the data into destination. I will add these detail in commit messages.